cmake/match.py

```diff
@@ -22,10 +22,17 @@
     input_lines = input.readlines()
     match_lines = match.readlines()
 
-if len(match_lines) != len(input_lines):
-    sys.exit(f"Line count doesn't match (is: {len(input_lines)}, expected: {len(match_lines)})")
+if len(match_lines) < len(input_lines):
+    sys.exit(f"Match length < input length (input: {len(input_lines)}, match: {len(match_lines)})")
 
+input_idx = 0
+optional_line = False
 for i, match_line in enumerate(match_lines):
+    opt = "${OPT}"
```

Comment:
hm, `{{OPT}}` to be consistent with how regex is specified?

Reply:
Not quite. I think Piotr suggested without `$` - to be in line with what you have a few lines below, in the code:
`# split into parts at {{ }}`